mirdata/rwc_popular.py

```diff
@@ -0,0 +1,374 @@
+"""RWC Popular Dataset Loader
+"""
+import csv
+import numpy as np
+import os
+
+import mirdata.utils as utils
+
+INDEX = utils.load_json_index("rwc_popular_index.json")
+METADATA = None
+METADATA_REMOTE = utils.RemoteFileMetadata(
+    filename='rwc-p.csv',
+    url='https://github.com/magdalenafuentes/metadata/archive/master.zip',
+    checksum='d566a64aa4bfb3e31004153b71348646')
+DATASET_DIR = 'RWC-Popular'
+ANNOTATIONS_REMOTE_1 = utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-P-2001.BEAT.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-P-2001.BEAT.zip',
+    checksum='3858aa989535bd7196b3cd07b512b5b6'
+)
+ANNOTATIONS_REMOTE_2 = utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-P-2001.CHORUS.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-P-2001.CHORUS.zip',
+    checksum='f76b3a32701fbd9bf78baa608f692a77'
+)
+ANNOTATIONS_REMOTE_3 = utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-P-2001.CHORD.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-P-2001.CHORD.zip',
+    checksum='68379c88bc8ec3f1907b32a3579197c5'
+)
+ANNOTATIONS_REMOTE_4 = utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-P-2001.VOCA_INST.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-P-2001.VOCA_INST.zip',
+    checksum='47ded648a496407ef49dba9c8bf80e87'
+)
+
+
+class Track(object):
+    def __init__(self, track_id, data_home=None):
+        if track_id not in INDEX:
+            raise ValueError('{} is not a valid track ID in RWC-Popular'.format(track_id))
+
+        self.track_id = track_id
+        self._data_home = data_home
+        self._track_paths = INDEX[track_id]
+
+        if METADATA is None or METADATA['data_home'] != data_home:
+            _reload_metadata(data_home)
+
+        if track_id in METADATA.keys():
+            self._track_metadata = METADATA[track_id]
+        else:
+            # annotations with missing metadata
```

Comment:
just to check - are there tracks with missing metadata in RWC? or is there complete metadata and this code is just reused from the Salami loader?

Reply:
Missing a tiny bit